### PR TITLE
Balance Tweaks Per Recent Test Results

### PR DESCRIPTION
**Pull request by Surrealaser**
:cl: Surrealistk
add: General circuits that can be adapted to a fire alarm, airlock or APC circuit with a multitool.
tweak: Damage instances from acid puddles capped at 1 / second.
tweak: Extended Barrel and RDS made slightly better for unwielded weapons. EB very slightly increases projectile speed.
tweak: Suppressor very slightly decreases projectile speed, but its accuracy bonus has slightly increased for wielded and unwielded weapons.
tweak: Laser Sight and Gyroscope made slightly better for wielded weapons (their benefits apply to them albeit at a lesser extent).
tweak: Barrel Charger slightly increases projectile speed.
tweak: Accuracy penalties for BFA, wielded and unwielded, slightly reduced.
tweak: M41A MK2 can now attach a recoil compensator.
tweak: Extra buckshot spread reduced; max range increased to 10 tiles. This should make investing in buckshot accuracy worth a damn as it can be at least somewhat effective out to mid range now.
tweak: Flechette spread slightly reduced as values were higher than intended per spread inherent to the shotgun.
tweak: Supply packs updated: orders of 2x standard guns now come with 2 AP mags of the appropriate type. 1x HPR + ammo cost reduced to 20 points, and valu pack of 5 HPR + ammo costs 80 points.
tweak: Neurotoxin projectile fire delays increased by +0.5 seconds across the board and progression smoothed out. Max range decreased to 10 tiles down from 15, and accurate range threshold decreased to 5 tiles down from 15, reducing its ability to off-screen; should still be very consistent at mid range.
/:cl:

### CODE REVIEW

File: tgmc.dme [hunk 1]
```diff
@@ -542,7 +542,6 @@
 #include "code\game\objects\items\books\book.dm"
 #include "code\game\objects\items\books\manuals.dm"
 #include "code\game\objects\items\circuitboards\AI_modules.dm"
-#include "code\game\objects\items\circuitboards\airlock.dm"
```
Comment (spookydonut) [inline tgmc.dme]: accident i think

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -434,7 +435,14 @@
 				else
 					user.visible_message("<span class='notice'>[user] removes [src]'s power control board.</span>",
 					"<span class='notice'>You remove [src]'s power control board.</span>")
-					new /obj/item/circuitboard/apc(loc)
+					var/obj/item/circuitboard/apc/circuit
```
Comment (spookydonut) [inline code/modules/power/apc.dm]: another accidental inclusion?

**Comment by DominikPanic** [comment]
Check spooky's comments.

File: code/modules/power/apc.dm (hunk 2 above)
Comment (Surrealaser) [reply code/modules/power/apc.dm]: No, this is intended.

File: tgmc.dme (hunk 1 above)
Comment (Surrealaser) [reply tgmc.dme]: Naw, consolidated the circuits cause they should be consolidated.

File: code/game/objects/items/circuitboards/circuitboard.dm [hunk 3]
```diff
@@ -64,3 +91,124 @@
...
+				t1 += "<a style='color: red' href='?src=\ref[src];access=[acc]'>[aname]</a><br>"
+
+	t1 += text("<p><a href='?src=\ref[];close=1'>Close</a></p>\n", src)
+
+	user << browse(t1, "window=airlock_electronics")
+	onclose(user, "airlock")
+
+
+/obj/item/circuitboard/airlock/Topic(href, href_list)
+	..()
```
Comment (DominikPanic) [inline code/game/objects/items/circuitboards/circuitboard.dm]:
```suggestion
	. = ..()
```

File: code/game/objects/items/circuitboards/circuitboard.dm [hunk 4]
```diff
@@ -64,3 +91,124 @@
...
+	if (usr.stat || usr.is_mob_restrained() || (!ishuman(usr) && !istype(usr,/mob/living/silicon)))
```
Comment (DominikPanic) [inline code/game/objects/items/circuitboards/circuitboard.dm]: I think there's an issynth or issilicon

File: code/game/objects/items/circuitboards/circuitboard.dm [hunk 5]
```diff
@@ -64,3 +91,124 @@
...
+		else
+			var/obj/item/I = usr.get_active_hand()
+			if (istype(I, /obj/item/device/pda))
+				var/obj/item/device/pda/pda = I
+				I = pda.id
+			if (I && src.check_access(I))
+				src.locked = 0
+				src.last_configurator = I:registered_name
+
+	if (locked)
```
Comment (DominikPanic) [inline code/game/objects/items/circuitboards/circuitboard.dm]:
```suggestion
	if(locked)
```
Also remove the whitespace on other if checks here.

File: code/game/objects/items/circuitboards/circuitboard.dm [hunk 6]
```diff
@@ -64,3 +91,124 @@
...
+	req_access = list(ACCESS_CIVILIAN_ENGINEERING)
+	var/list/conf_access = null
+	var/one_access = 0 //if set to 1, door would receive req_one_access instead of req_access
+	var/last_configurator = null
+	var/locked = 1
+
+
+/obj/item/circuitboard/airlock/attack_self(mob/user as mob)
+	if (!ishuman(user) && !istype(user,/mob/living/silicon/robot))
+		return ..(user)
```
Comment (DominikPanic) [inline code/game/objects/items/circuitboards/circuitboard.dm]: `..()` passes arguments by default, therefore you should remove the user there.

File: code/game/objects/items/circuitboards/circuitboard.dm [hunk 7]
```diff
@@ -8,10 +8,36 @@
...
+				new_board = new/obj/item/circuitboard/airalarm(user.loc)
+			if("None")
+				to_chat(user, "<span class='notice'>You choose not to alter the function of [name] at this time.</span>")
+		if(new_board)
+			qdel(src)
+			to_chat(user, "<span class='notice'>You set the general circuit to act as [name].</span>")
+			new_board.set_general()
+			user.put_in_hands(new_board)
+		return
+	. = ..()
```
Comment (DominikPanic) [inline code/game/objects/items/circuitboards/circuitboard.dm]:
```suggestion
	return ..()
```

**Comment by DominikPanic** [comment]
Also conflicts.

**Pull request merged**